Repo main

neutralboolean opened the pull request.

neutralboolean committed:
Stubbed services and controller

neutralboolean committed:
GSON test of deserialization

Pushing to make changes current across multiple clones

neutralboolean committed:
Preliminary conversion to Jackson

Brought Jackson usage to par with previous GSON implementation

neutralboolean committed:
Finished Jackson integration

-modified MongoDocument and removed 'volume' property/field
+completed 'days'-limited results output

neutralboolean committed:
Safe deletion: JMS

neutralboolean committed:
Safe delete: Settlement/Translation

neutralboolean committed:
Safe delete: IEX and Settlements

neutralboolean committed:
dang, still works

neutralboolean committed:
Final safe deletion

neutralboolean committed:
mirroring changes in develop branch

neutralboolean committed:
Update StockPriceController.java

neutralboolean committed:
Embedded custom serializer for MongoDocument

neutralboolean committed:
prepping necessary files for the future

going off of feedback, adding in some files that might be necessary later.

neutralboolean committed:
polish off initial testing

neutralboolean committed:
Update StockPriceControllerTest.java

defined proper ResultMatcher to complete tests

neutralboolean committed:
Update StockPriceControllerTest.java

cleaning out imports

neutralboolean committed:
Adding RestTemplate Bean to Config

neutralboolean committed:
Making Mongo POJO changes current

neutralboolean committed:
Making Serialization current

neutralboolean committed:
Update Service and Controller

neutralboolean committed:
Cleaned up service pathway for StockPriceController

Properly separated StockPriceController from AlphaVantageService to prepare for support of "internal" data repository. PriceRequestService is now the gateway to either internal or external request.

neutralboolean committed:
Modified `date` property of MongoDocument

Converted `date` from String to Date.class and modified affected files to support the change.

neutralboolean committed:
Type conversion of `date` in MongoDocument.class

Type of `date` property switched to Instant to support zone switching in future. Modified related files to smooth conversion.

neutralboolean committed:
brought in more suggested fuse dependencies

neutralboolean committed:
Merge branch 'develop' into annihilation

neutralboolean committed:
Merge pull request #1 from neutralboolean/annihilation

Annihilation

neutralboolean committed:
Update AlphaVantageService.java

Cleaning up some mistakes from merging

neutralboolean committed:
Style corrections

neutralboolean committed:
Update MongoDocument.java

neutralboolean committed:
Pruning back MongoDB files for new branch

neutralboolean committed:
Began implementation of MongoDB support

it's borked right now

neutralboolean committed:
Updated dependencies for spring-data-mongodb

Turns out the spring-data-mongo has new Java driver dependencies, it seems. Saving work for remote.

mikegajda commented on src/main/java/org/galatea/starter/domain/MongoDocument.java:
```diff
@@ -0,0 +1,28 @@
+@JsonSerialize(using = MongoDocSerializer.class)
+@NoArgsConstructor
+@Document
+@Data public class MongoDocument {
```
Consider naming this something closer to the function it's serving, like StockTick or StockData, or something similar.

mikegajda commented on src/main/java/org/galatea/starter/entrypoint/StockPriceController.java:
```diff
@@ -0,0 +1,65 @@
+    log.info("Returned from PriceRequestService with map of MongoDocuments.");
+    assert processed != null && !processed.isEmpty(); //this might not be assertable in the future
+    // constructing the array of 'days'-limited stock price results
+    ObjectMapper objectMapper = new ObjectMapper();
```
Consider avoiding your own JSON creation by directly using ObjectMapper. For REST endpoints, you can return any serializable object (so any object that Jackson knows how to serialize by default, or any object that you've marked up with JsonProperties) and Jackson will handle the conversion for you. What about changing the return type of this function to `TreeMap<String, MongoDocument>` and simply returning `processed`?

mikegajda commented on src/main/java/org/galatea/starter/service/AlphaVantageService.java:
```diff
@@ -0,0 +1,92 @@
+        = "https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&symbol=" + symbol;
+    String output = days > 100 ? "full" : "compact";
+    String requestUrl = alphaVantageUrl + "&outputsize=" + output + "&apikey=" + MyProps.apiKey;
+    ResponseEntity<String> response = restTemplate.getForEntity(requestUrl, String.class);
```
Consider mapping this to an intermediate object, or directly to MongoDocument instead of getting a String.class and then doing the conversion manually with mapJsonGraph()

neutralboolean committed:
caching changes

neutralboolean committed:
changes to fix MongoConfig issues

neutralboolean committed:
finalized MongoDB implementation

~modified returns to from Maps to Lists
~modified logs to update terms
-removed some old, commented functionality

neutralboolean committed:
test stub & name refactors

neutralboolean committed:
patched oversight in request logic

PriceRequestService didn't properly check dates before making requests. So that's fixed now. Yup.

cbaltera commented:
Closing this as part of the intro project
